### PR TITLE
Add prefixes to layer names and layer output names

### LINKED ISSUE #21698: DNN/ONNX: outputs registration regression, feature request for new version of Clip operator
##### System information (version)

- OpenCV => 4.5.5
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2019

##### Detailed description

I tried to infer using a Selfie Segmenter ONNX model (you can find the model here: https://github.com/PINTO0309/PINTO_model_zoo/tree/main/109_Selfie_Segmentation), however I get Nan on all output values.

##### Steps to reproduce

You can replicate this issue simply by running this simple script with OpenCV 4.5.5:

```
#include <iostream>

#include <opencv2/imgcodecs.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/dnn.hpp>
#include <opencv2/core.hpp>

int main()
{
    cv::Size inputSizeNewBarracuda = cv::Size(256, 256);

    std::string imagefilename = "C:/Lixo/SantaNoel.jpg";
    std::string newBarracuda = "C:/Users/cesar.gouveia/Downloads/saved_model_openvino/model_float32.onnx";

    cv::dnn::Net net = cv::dnn::readNetFromONNX(newBarracuda);
    net.setPreferableBackend(cv::dnn::DNN_BACKEND_OPENCV);
    net.setPreferableTarget(cv::dnn::DNN_TARGET_CPU);

    cv::Mat img = cv::imread(imagefilename);
    cv::Mat resized;
    cv::resize(img, resized, inputSizeNewBarracuda);

    std::vector<cv::Mat> imgBatch = { resized };
    bool swapRBChannels = false;
    cv::Mat blob = cv::dnn::blobFromImages(imgBatch, 1.0, cv::Size(), cv::Scalar(), swapRBChannels, false, CV_32F);
    blob = blob.reshape(1, { 1, inputSizeNewBarracuda.height, inputSizeNewBarracuda.width, 3 }); // because the model has input in channels last

    net.setInput(blob);

    std::vector<cv::Mat> outputs;
    outputs.clear();

    std::vector<cv::String> unconnectedOutLayerNames = net.getUnconnectedOutLayersNames();
    net.forward(outputs, unconnectedOutLayerNames);

    const cv::Mat& targetMat = outputs[0];
    const float* targetBuffer = (float*)targetMat.data;

    std::cout << targetMat.size[0] << std::endl; // 1
    std::cout << targetMat.size[1] << std::endl; // 256
    std::cout << targetMat.size[2] << std::endl; // 256
    std::cout << targetMat.size[3] << std::endl; // 1

    // Access contiguos buffer
    for (size_t i = 0; i < 256 * 256; i++)
    {
        std::cout << targetBuffer[i] << std::endl;
    }

    // Access Mat
    targetMat.reshape(1, { 1, 256, 256 });
    for (size_t o = 0; o < targetMat.size[0]; o++)
    {
        for (size_t i = 0; i < targetMat.size[1]; i++)
        {
            for (size_t j = 0; j < outputs[0].size[2]; j++)
            {
                std::cout << outputs[0].at<float>(o, i, j);
            }
        }
    }
    
    std::cout << "Finished" << std::endl;
}
```

The model can be downloaded by the link I provided. This is the first "channels last" model that I use with OpenCVDNN, all my other models are channel first and I never had this code behavior before. I tried to access the mat and the contiguos memory array but neither of them worked. 

Thanks,
César.

### PR DESCRIPTION
**Pull request by rogday**
**Merge with extra:** https://github.com/opencv/opencv_extra/pull/968
Fixes #21698

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [X] There is a reference to the original bug report and related work
- [X] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -723,10 +723,11 @@ void ONNXImporter::populateNet()
...
-    if (node_proto.has_name() && !node_proto.name().empty())
-    {
-        return node_proto.name();
-    }
+//    We need to rework DNN outputs API first, this causes #21698 right now
+//    if (node_proto.has_name() && !node_proto.name().empty())
+//    {
+//        return node_proto.name();
+//    }
     for (int i = 0; i < node_proto.output_size(); ++i)
```
Comment (alalek) [inline modules/dnn/src/onnx/onnx_importer.cpp:731]: Problem is still here.
There are several "outputs" with different names. Why do we pick the first non-empty output name?

IMHO, we should register all "outputs" instead.

Comment (alalek) [reply modules/dnn/src/onnx/onnx_importer.cpp:731]: @rogday Could you please prepare a small test case for this problem? I will try to fix it

Comment (rogday) [reply modules/dnn/src/onnx/onnx_importer.cpp:731]: @alalek, sure, you could use tests at https://github.com/opencv/opencv_extra/pull/968.

**Comment by asmorkalov** [comment]
@zihaomu Could you review the patch and comment.

**Review by zihaomu — APPROVED** [review]
LGTM! :+1:

**Comment by asmorkalov** [comment]
@rogday 
```
[  FAILED  ] Test_ONNX_layers.OutputRegistration/0, where GetParam() = OCV/CPU
```

**Review by zihaomu — APPROVED** [review]

**Review by alalek — APPROVED** [review]

**Pull request merged**